subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstylePlugin.groovy

```diff
@@ -57,6 +58,7 @@ class CheckstylePlugin extends AbstractCodeQualityPlugin<Checkstyle> {
             configProperties = { extension.configProperties }
             ignoreFailures = { extension.ignoreFailures }
             showViolations = { extension.showViolations }
+            antClassName = { getAntClassForToolVersion(extension.toolVersion) }
```

Comment:
Yes, that's quite possibly the best way to go. It's unfortunate that you'll have to construct a classloader for the checkstyle configuration just to detect this, before getting Ant to load the classes itself. It would be good to avoid this: I'm not super-familiar with the Ant integration, so you might want to ask on the dev list if there's not a clear way forward. Thanks!